packages/fxa-payments-server/src/routes/Product/PayPalButton/index.tsx

```diff
@@ -1,19 +1,89 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import ReactDOM from 'react-dom';
 
+import * as apiClient from '../../../lib/apiClient';
+import { Customer } from '../../../store/types';
+import { SubscriptionCreateAuthServerAPIs } from '../SubscriptionCreate';
+
 declare var paypal: {
   Buttons: {
     driver: Function;
   };
 };
 
-const PaypalButtonBase = paypal.Buttons.driver('react', {
-  React,
-  ReactDOM,
-});
+export type PaypalButtonProps = {
+  apiClientOverrides?: Partial<SubscriptionCreateAuthServerAPIs>;
+  customer: Customer | null;
+  setPaymentError: Function;
+  idempotencyKey: string;
+  ButtonBase?: React.ElementType;
+};
+
+export type ButtonBaseProps = {
+  createOrder?: (
+    event: React.MouseEvent<HTMLButtonElement, globalThis.MouseEvent>
+  ) => void;
+  onError?: (
+    event: React.MouseEvent<HTMLButtonElement, globalThis.MouseEvent>
+  ) => void;
+};
+
+export const PaypalButtonBase =
+  typeof paypal !== 'undefined'
+    ? paypal.Buttons.driver('react', {
+        React,
+        ReactDOM,
+      })
+    : null;
+
+export const PaypalButton = ({
+  apiClientOverrides,
+  customer,
+  setPaymentError,
+  idempotencyKey,
+  ButtonBase = PaypalButtonBase,
+}: PaypalButtonProps) => {
+  const createOrder = useCallback(async () => {
+    try {
+      const { apiCreateCustomer, apiGetPaypalCheckoutToken } = {
+        ...apiClient,
+        ...apiClientOverrides,
+      };
+      if (!customer) {
+        await apiCreateCustomer({
+          idempotencyKey,
+        });
+      }
+      const { token } = await apiGetPaypalCheckoutToken();
+      return token;
+    } catch (error) {
+      console.error('onCreateOrder failed', error);
```

Comment:
Remove?

Reply:
I copied this from the [catch block](https://github.com/mozilla/fxa/blob/deace7808044003eca11c5bd67c1cf17aa9ac262/packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx#L136-L139) of `handleSubscriptionPayment` for Stripe form submission. Should that also be removed? I agree it's not all that helpful in production.

Reply:
Is it helpful to have those errors reported to Sentry?  We can leave them if they help with monitor or debugging.

Reply:
I removed the new ones I added. I don't think they're that helpful.